engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala

```diff
@@ -24,52 +23,31 @@ object TestReporter {
 
   private val instances: ConcurrentHashMap[String, TestReporter] = new ConcurrentHashMap[String, TestReporter]()
 
-  def resetAll(): Unit =
-    instances.values().asScala.foreach(_.reset())
-
-  def reset(name: String): Unit =
-    get(name).reset()
-
-  def removeAll: Unit =
-    instances.clear()
-
   def remove(name: String): Unit =
     instances.remove(name)
 
-  def append(name: String, reporter: TestReporter): Unit =
+  def add(name: String, reporter: TestReporter): Unit =
     instances.put(name, reporter)
 
-  def get(name: String): TestReporter =
-    Option(
-      TestReporter
-      .instances
-      .get(name)
-    ).getOrElse(
-      throw new IllegalArgumentException("Reporter doesn't exists.")
-    )
+  def get(name: String): TestReporter = TestReporter.instances.asScala
+    .getOrElse(name, throw new IllegalArgumentException("Reporter doesn't exists."))
+
 }
 
-class TestReporter extends AbstractReporter {
+class TestReporter extends MetricReporter with CharacterFilter {
 
-  def reset(): Unit = {
-    histograms.clear()
-    gauges.clear()
-    counters.clear()
-    meters.clear()
-  }
+  private val processToMetric = new ConcurrentHashMap[(ProcessName, String), Metric]()
 
-  def testHistograms: Map[Histogram, String] = histograms.asScala.toMap
+  def testMetrics[T<:Metric](metricNamePattern: String)(implicit scenarioName: ProcessName): Iterable[T] =
+    processToMetric.asScala.filterKeys { case (scenarioName1, metricName1) =>
+      scenarioName1 == scenarioName && metricName1.contains(metricNamePattern)
```

Comment:
So after looking in the code, I see that there was two problem:
- we returned metrics for other concurrent scenarios
- we tested instant rate 
Have u reproduced this bugs?

Reply:
TBH, I tried mulitple times locally and I haven't succeeded... We have to try it out on CI...